src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1433,6 +1445,8 @@ public int Receive(byte[] buffer, int offset, int size, SocketFlags socketFlags,
             int bytesTransferred;
             errorCode = SocketPal.Receive(_handle, buffer, offset, size, socketFlags, out bytesTransferred);
 
+            errorCode = UpdateReceiveSocketErrorForCleanedUp(errorCode, bytesTransferred);
```

Comment:
Why do we do this even if errorCode == Success?

Same for the other such occurrences below.

Reply:
On Linux, when the UDP receive is unblocked `bytesTransferred` is `zero` and `errorCode` is `Success`.
I can add a comment about this in UpdateReceiveSocketErrorForCleanedUp.